### ISSUE TITLE
Creating JLine reader preemptively reads input stream 

### ISSUE THREAD

**Issue by bdw429s** [jline/jline3#383]
When I create the reader instance, any input on my standard input is disappearing, presumable read in by the reader preemptively even though I haven't called `readLine()`.  This causes undesirable behavior when the std in contains text which is NOT meant for my app in situations where my console app exits prior to ever running `readLine()`.  

Take this example.  `box` is the name of the my java-based JLine-powered console app. If I copy all of the following to my clipboard and paste it into Windows  `cmd`:
```bash
cd c:\sandbox
box ver
echo blah
```
In this case, even though my app creates a JLine instance, it sees that there are args passed in so it enters command mode, and if the command running declines to read from the standard input, then my command exits which should still leave `echo blah` unread on the standard input which will get passed to `cmd` as a third command.  Instead, that text simply disappears.  

The TL;DR; of this is JLine should not read anything from the standard input until the first time I actually call `readLine()`.  Until then, the standard input should remain untouched in case I decide not to read it at all.

**Comment by bdw429s** [comment]
Related CommandBox ticket: https://ortussolutions.atlassian.net/browse/COMMANDBOX-975

**Comment by gnodet** [comment]
The behavior is controlled with the `pause`/`resume` calls on the `Terminal` interface: 
 https://github.com/jline/jline3/blob/master/terminal/src/main/java/org/jline/terminal/Terminal.java#L122-L171

You can also set the initial state on the `TerminalBuilder` using
  https://github.com/jline/jline3/blob/master/terminal/src/main/java/org/jline/terminal/TerminalBuilder.java#L243-L257

This is all related to https://github.com/jline/jline3/issues/266

**Comment by bdw429s** [comment]
> The behavior is controlled with the pause/resume calls on the Terminal interface:

@gnodet This suggestion does not work.  Please see this bug that prevents it from being usable:

https://github.com/jline/jline3/issues/462
